cypress-io/cypress
# chore: add batch execution to CloudDataSource

### Additional details
Addresses @flotwig comment on #21250 re: network queries, batching multiple GraphQL requests into a single remote request with rewritten variables, using `@graphql-tools/batch-execute`. This makes it simpler to batch on the remote server.

The documentation here explains how the merging works: https://www.graphql-tools.com/docs/batch-execution

Additionally only does this in situations where we have multiple requests to batch, so we don't end up with rewritten queries on every request, but just the ones where we have multiple requests. Limits the batching to max 20 items per-query

### Steps to test
- Checkout the branch
- Observe fewer network requests

### How has the user experience changed?

Before:


https://user-images.githubusercontent.com/154748/175098776-fc16a10a-eb3b-4ebd-b126-7f9652f3c9fd.mp4



After:

https://user-images.githubusercontent.com/154748/175099329-66035f11-0b70-4d9a-adab-8bfe04991c55.mp4



### PR Tasks
